libraries/botbuilder/src/activityValidator.ts

```diff
@@ -0,0 +1,27 @@
+/**
+ * @module botbuilder
+ */
+/**
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { Activity } from 'botbuilder-core';
+
+export function validateActivity(resolve: any, activity: Activity): void{
```

Comment:
I'd prefer if the order of the parameters was switched, so instead of 
```ts
export function validateActivity(resolve: any, activity: Activity): void{
```

it would be
```ts
export function validateActivity(activity: Activity, resolve: any): void {
```

The Activity is the main bit we care about, and the resolve is something we use because of the original composition of the code.

Reply:
Actually, can you instead just have this function return an `Activity` instead of void, so in the calling code it looks like this

```ts
if (req.body) {
    try {
        const activity = validateActivity(req.body);
        resolve(activity);
    } catch (err) {
        reject(err);
    }
} else {
    let requestData = '';
    req.on('data', (chunk: string): void => {
        requestData += chunk;
    });
    req.on('end', (): void => {
        try {
            req.body = JSON.parse(requestData);
            const activity = validateActivity(req.body);
            resolve(activity);
        } catch (err) {
            reject(err);
        }
    });
}
```